extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java

```diff
@@ -200,6 +200,11 @@ public void build(ApplicationInfoBuildItem applicationInfo,
                             resourceEntry.getValue().getBytes(StandardCharsets.UTF_8)));
         }
 
+        try {
+            Files.delete(root);
```

Comment:
`root` is a directory, you need to recurse into its subfolders to delete it

Reply:
I think there is an utility method somewhere

Reply:
`io.quarkus.deployment.util.FileUtil#deleteDirectory`

Reply:
Actually we don't because the contents have already been deleted by Dekorate. But Dekorate doesn't delete the directory itself, which is what this commit addresses

Reply:
or `io.quarkus.bootstrap.util.IoUtils#recursiveDelete`

Reply:
Note: we should really merge similar utility methods someday :)

Reply:
In any case, I added it and made it a little more foolproof